### PR TITLE
[BE] Enable ruff's UP rules and autoformat optim/

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #105437
* #105436
* #105435
* #105434
* #105433
* #105432
* #105431
* #105430
* #105429
* #105428
* #105427
* __->__ #105426
* #105425
* #105424
* #105423

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Review by albanD — APPROVED** [review]

**Review by aaronenyeshi — APPROVED** [review]
profiler changes LGTM, please take a look at lint issues

File: torch/distributions/mixture_same_family.py [hunk 1]
```diff
@@ -71,17 +71,17 @@ def __init__(self,
         cdbs = self._component_distribution.batch_shape[:-1]
         for size1, size2 in zip(reversed(mdbs), reversed(cdbs)):
             if size1 != 1 and size2 != 1 and size1 != size2:
-                raise ValueError("`mixture_distribution.batch_shape` ({0}) is not "
+                raise ValueError("`mixture_distribution.batch_shape` ({}) is not "
```
Comment (janeyx99) [inline torch/distributions/mixture_same_family.py:74]: why not use an fstring here?

**Review by janeyx99 — APPROVED** [review]

File: torch/distributions/mixture_same_family.py (hunk 1 above)
Comment (justinchuby) [reply torch/distributions/mixture_same_family.py:74]: done

File: torch/distributions/mixture_same_family.py [hunk 2]
```diff
@@ -71,17 +71,17 @@ def __init__(self,
...
                                  "compatible with `component_distribution."
-                                 "batch_shape`({1})".format(mdbs, cdbs))
+                                 "batch_shape`({})".format(mdbs, cdbs))
 
         # Check that the number of mixture component matches
         km = self._mixture_distribution.logits.shape[-1]
         kc = self._component_distribution.batch_shape[-1]
         if km is not None and kc is not None and km != kc:
-            raise ValueError("`mixture_distribution component` ({0}) does not"
+            raise ValueError("`mixture_distribution component` ({}) does not"
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply torch/distributions/mixture_same_family.py:82]: Done

**Pull request closed without merging**